trader/trader.go

```diff
@@ -117,6 +121,13 @@ func (t *Trader) Start() {
 		currentUpdateTime := time.Now()
 		if lastUpdateTime.IsZero() || t.timeController.ShouldUpdate(lastUpdateTime, currentUpdateTime) {
 			success := t.update()
+			t.threadTracker.TriggerGoroutine(func(inputs []interface{}) {
+				e := t.metricsTracker.SendUpdateEvent(currentUpdateTime, success)
+				if e != nil {
+					log.Printf("failed to send update event metric: %s", e)
```

Comment:
TODO NS: comment why we are not throwing an error here

Reply:
Again, I think this conflicts with [this](https://github.com/stellar/kelp/pull/513#discussion_r500364496). I'm throwing a `log.Fatalf` around the error now, since that's the pattern used for other errors in the `deleteAllOffers` function.

Reply:
we want to check the error produced, which has been added, but we want to figure out what to do with the error. Also there are two errors, once in the context of the outer code, the other in the context of the inner goroutine function. Not sure if that's the conflict you were referring to (the link seems broken for me so I'm not sure)

We do not want to do a `log.Fatalf` because the error stems from a failure to send metrics. Logging with a `log.Printf` is the right thing to do. This is because of it being a metric more than it being from a goroutine. We don't want to crash the bot on metric issues (except at startup temporarily, which we do for now).